Counterexamples/SeparableNotSecondCountable.lean

```diff
@@ -0,0 +1,51 @@
+/-
+Copyright (c) 2025 Yury G. Kudryashov. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Yury G. Kudryashov
+-/
+import Mathlib.Analysis.Real.Cardinality
+
+/-!
+# Example of a linear order which is a separable space but is not a second countable topology
+
+In this file we provide an example of a linear order
+such that the order topology is a separable space but is not a second countable topology.
+
+The example is `ℝ ×ₗ Bool` which is the real line with each point duplicated
+so that the duplicate is greater than the original point
+and points with different real values are compared by these values.
+-/
+
+open Set TopologicalSpace
+
+namespace RealProdLexBool
+
+instance instTopologicalSpace : TopologicalSpace (ℝ ×ₗ Bool) := Preorder.topology _
+instance instOrderTopology : OrderTopology (ℝ ×ₗ Bool) := ⟨rfl⟩
+
```

Comment:
maybe add `noncomputable instance : LinearOrder (ℝ ×ₗ Bool) := by infer_instance` to confirm that this is indeed a linear order?